zktraffic/endpoints/stats_server.py

```diff
@@ -77,27 +79,50 @@ def wakeup(self):
   def has_stats(self):
     return len(self._get_stats('per_path')) > 0
 
-  def _get_stats(self, name, prefix=''):
+  def _get_stats(self, name, prefix='', output_array=False):
     stats_by_opname = self._stats.stats(name, self._max_results)
 
-    stats = {}
+    if output_array:
+        stats_arr = []
+    else:
+        stats = {}
     for opname, opstats in stats_by_opname.items():
       for path, value in opstats.items():
-        stats["%s%s%s" % (prefix, opname, path)] = value
-
-    return stats
+        if output_array:
+            #TODO: split prefix to each own key
+            #TODO: when per_ip: split path by ':' and add IPs to their own key
+            if prefix.endswith('/'):
+                prefix=prefix[:-1]
+            tmp_dict={"opname": opname, "path": "%s%s" % (prefix, path), "value":value}
+            stats_arr.append(tmp_dict)
+        else:
+            stats["%s%s%s" % (prefix, opname, path)] = value
+    if output_array:
+        stats_json=json.dumps(stats_arr)
+        return stats_json
+    else:
+        return stats
 
   @HttpServer.route("/json/paths")
-  def json_paths(self):
-    return self._get_stats('per_path')
+  def json_paths(self, array_output=False):
```

Comment:
I wonder if you could move `array_output` inside the method body, so that it still can be used internally, but it won't then change the method signature. What do you think?

You could also shorten it to, for example:

```
array_output = array_get and array_get.lower() in ['true', '1', 't', 'y', 'yes']
```

And if the `array_get` is simply an empty string when there is no `GET` parameter passed, then it can be made shorter even more.

What do you think?

Reply:
This is taken care of now as well. Thanks!